api/uniter/unit.go

```diff
@@ -644,13 +644,14 @@ func (u *Unit) RequestReboot() error {
 	return result.OneError()
 }
 
-// JoinedRelations returns the tags of the relations the unit has joined.
-func (u *Unit) JoinedRelations() ([]names.RelationTag, error) {
+// RelationsInScopeOrSuspended returns the tags of the relations the unit has joined
```

Comment:
so when a relation is suspended, it's no longer in scope? this is starting to get pretty messy, and if you're talking about making "error" also count towards this, it gets even less clear. can it not be in scope, but also suspended? and have the uniter filter suspended relations out as appropriate

Reply:
When a relation is suspended, it has to leave scope and have the departed / broken hooks run.
What we really care about is should the relation state dir be removed when the uniter initialises. If a relation is not in scope, but could be again, the state dir won't be empty and so we shouldn't delete it. So I guess the concept is "active" vs "non-active" relations rather than "in scope" vs "out of scope". We make a single api call off unit which only returns tags - I guess a new bespoke call could be done that returns tag,inscope,status

Reply:
OK, sounds fair, but please come up with a name that describes that, and not "isXorYorZ". Not sure about (non-)active, because something that's non-active can become active again.

Reply:
I've added a new uniter api call "RelationStatus" which returns the scope and status info as needed by the uniter.